ext/BijectorsEnzymeCoreExt.jl

```diff
@@ -0,0 +1,184 @@
+        b isa Union{Const,Duplicated}
+    ) && !(
+        RT <: Union{Const,BatchDuplicated,BatchDuplicatedNoNeed} &&
+        wt_y isa Union{Const,BatchDuplicated} &&
+        wt_u_hat isa Union{Const,BatchDuplicated} &&
+        b isa Union{Const,BatchDuplicated}
+    )
+        throw(ArgumentError("inconsistent activities"))
+    end
+
+    # Compute primal value
+    Ω = find_alpha(wt_y.val, wt_u_hat.val, b.val)
+
+    # Early exit if no derivatives are requested
+    if RT <: Const
+        return Ω
+    end
+
+    Ω̇ = if wt_y isa Const && wt_u_hat isa Const && b isa Const
+        # Trivial case: All partial derivatives are 0
+        zero(Ω)
+    else
+        # In all other cases we have to compute the partial derivatives
+        ∂Ω_∂wt_y, ∂Ω_∂wt_u_hat, ∂Ω_∂b = ∂find_alpha(Ω, wt_y, wt_u_hat, b)
+        _muladd_partial(
+            ∂Ω_∂wt_y,
+            wt_y,
+            _muladd_partial(∂Ω_∂wt_u_hat, wt_u_hat, _muladd_partial(∂Ω_∂b, b, nothing)),
+        )
+    end
+
+    if RT <: Duplicated
+        @assert Ω̇ isa Real
+        return Duplicated(Ω, Ω̇)
+    elseif RT <: DuplicatedNoNeed
+        @assert Ω̇ isa Real
+        return Ω̇
+    elseif RT <: BatchDuplicated
+        @assert Ω̇ isa Tuple{Vararg{Real}}
+        return BatchDuplicated(Ω, Ω̇)
+    else
+        @assert RT <: BatchDuplicatedNoNeed
+        @assert Ω̇ isa Tuple{Vararg{Real}}
+        return Ω̇
+    end
+end
+
+function EnzymeRules.augmented_primal(
+    config::EnzymeRules.Config,
+    ::Const{typeof(find_alpha)},
+    ::Type{RT},
+    wt_y::Union{Const,Active},
+    wt_u_hat::Union{Const,Active},
+    b::Union{Const,Active},
+) where {RT<:Union{Const,Active}}
+    # Only compute the the original return value if it is actually needed
+    Ω =
+        if EnzymeRules.needs_primal(config) ||
+            EnzymeRules.needs_shadow(config) ||
+            (RT <: Active && !(wt_y isa Const && wt_u_hat isa Const && b isa Const))
+            find_alpha(wt_y.val, wt_u_hat.val, b.val)
+        else
+            nothing
+        end
+
+    tape = if RT <: Const || (wt_y isa Const && wt_u_hat isa Const && b isa Const)
+        # Trivial case: No differentiation or all partial derivatives are 0
+        # Thus no tape is needed
+        nothing
+    else
+        # Store the partial derivatives
+        ∂find_alpha(Ω, wt_y, wt_u_hat, b)
+    end
+
+    # Ensure that we follow the interface requirements of `augmented_primal`
+    primal = EnzymeRules.needs_primal(config) ? Ω : nothing
+    shadow = EnzymeRules.needs_shadow(config) ? zero(Ω) : nothing
+
+    return EnzymeRules.AugmentedReturn(primal, shadow, tape)
+end
+
+function EnzymeRules.reverse(
+    config::EnzymeRules.ConfigWidth{1},
+    ::Const{typeof(find_alpha)},
+    ::Type{<:Const},
+    ::Nothing,
+    wt_y::Union{Const,Active},
+    wt_u_hat::Union{Const,Active},
+    b::Union{Const,Active},
+)
+    # Trivial case: Nothing to be differentiated (return activity is `Const`)
```

Comment:
These should also likely be extended for batch size != 1.

Basically in these cases active returns are just an ntuple of the relevant derivatives (in this case zero)

Reply:
I added support for batches in reverse mode (or at least I tried to 😄). How would I go about testing it? AFAICT `EnzymeTestUtils.test_reverse` won't test batched mode since all activities are either `Const` or `Active`.

Reply:
The easiest way is probably to make a wrapper function that takes and returns variables by reference then calls this function. Then you can use batchduplicated

Reply:
I tried to add such a test in https://github.com/TuringLang/Bijectors.jl/pull/349/commits/22471daa7829bec25ce25f9d03d06bcfd469f248 but now julia is aborted in CI due to an LLVM assertion error: https://github.com/TuringLang/Bijectors.jl/actions/runs/11788862392/job/32836666863?pr=349#step:5:488 Do you have any idea what is going and what the problem is?